libs/designsystem/src/lib/components/modal/modal-wrapper/modal-wrapper.component.ts

```diff
@@ -71,13 +74,17 @@ export class ModalWrapperComponent implements Modal, AfterViewInit, OnInit, OnDe
     private injector: Injector,
     private elementRef: ElementRef<HTMLElement>,
     private renderer: Renderer2,
-    private resizeObserverService: ResizeObserverService
+    private resizeObserverService: ResizeObserverService,
+    private windowRef: WindowRef
   ) {
     this.observeViewportResize();
   }
 
   ngOnInit(): void {
     this.ionModalElement = this.elementRef.nativeElement.closest('ion-modal');
+    this.embeddedComponentElement = this.ionContentElement.nativeElement.querySelector(
+      ':first-child'
+    );
```

Comment:
Consider moving to `ngAfterViewInit` (where we check for modal flavor) or maybe even inside `onEmbededComponentElementResize()`